hypothesis-python/tests/django/toystore/forms.py

```diff
@@ -0,0 +1,182 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis/
+#
+# Most of this work is copyright (C) 2013-2019 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import absolute_import, division, print_function
+
+from django import forms
+from django.forms import widgets
+
+from tests.django.toystore.models import (
+    CouldBeCharming,
+    Customer,
+    ManyNumerics,
+    ManyTimes,
+    OddFields,
+)
+
+
+class ReprModelForm(forms.ModelForm):
+    def __repr__(self):
+        """I recommend putting this in your form to show the failed cases."""
+        return "%r\n%r" % (self.data, self.errors)
+
+
+class ReprForm(forms.Form):
+    def __repr__(self):
+        return repr(self.data) + repr(self.errors)
+
+
+class CouldBeCharmingForm(ReprModelForm):
+    class Meta:
+        model = CouldBeCharming
+        fields = "__all__"
+
+
+class CustomerForm(ReprModelForm):
+    class Meta:
+        model = Customer
+        fields = "__all__"
+
+
+class ManyNumericsForm(ReprModelForm):
+    class Meta:
+        model = ManyNumerics
+        fields = "__all__"
+
+
+class ManyTimesForm(ReprModelForm):
+    class Meta:
+        model = ManyTimes
+        fields = "__all__"
+
+
+class OddFieldsForm(ReprModelForm):
+    class Meta:
+        model = OddFields
+        fields = "__all__"
+
+
+class DynamicForm(ReprForm):
+    def __init__(self, *args, field_count=5, **kwargs):
```

Comment:
Sadly keyword-only arguments are not valid syntax on Python 2, which we still test against 😭